### ISSUE TITLE
Closing and reopening a ride causes every peep to leave the queue

### ISSUE THREAD

**Issue by devnoname120** [OpenRCT2/OpenRCT2#2916]
If I close a ride and reopen it right after, peeps go all the way to the entrance anyway before leaving the queue, as if it were closed.

Instead, I think that peeps should not leave the queue, even if the ride is closed, but neither should new peeps enter it. If they wait too long, they will leave the queue as they would normally. This would avoid emptying the whole queue when doing a minor edit on a ride.
This feature might annoy some users though.

Anyway, I think that peeps should at least not leave the queue if a ride is reopened before they reach the entrance.

**Comment by duncanspumpkin** [comment]
I disagree with your first idea but there is no reason why the second idea couldn't be done.

**Comment by devnoname120** [comment]
For the first idea, the behavior could be triggered with Ctrl+Click, but it's not very useful, so it might not be worth doing it.

**Comment by IntelOrca** [comment]
It is indeed annoying for all guests to leave the queue if you close it (maybe even by accident or to quickly change a setting). However you have the problem with the game distinguishing whether you intended the ride to be closed and all guests to leave or not.

**Comment by devnoname120** [comment]
Well, that's what I had in mind with peeps leaving if they waited for too long, just like when rides get broken. I don't see any inconvenience except the stress to see peeps waiting and leaving if you are not quick enough to do the modification, and feeling that they are waiting uselessly if you undertake huge modifications, because they will leave anyway before you finish.

**Comment by ccfreak2k** [comment]
The only thing I can think of for implementing this in the UI, besides adding some kind of option, would be to default to not having the peeps leave the queue, but do make them leave if you double-click the closed button, which already also immediately removes the cars and dumps riders at the exit.

**Comment by Patrik356b** [comment]
One thing that comes to mind is to: pause, close, edit, open, unpause 
And the guests should still be queuing like nothing happened
